### ISSUE TITLE
Accessibility of scrollable Tabbar

### ISSUE THREAD

**Issue by kynara** [flutter/flutter#177309]
### Steps to reproduce

1. Create a TabBar similar to this and run it on a mobile device. (See example code)
2. Enter Talkback mode and try to swipe through the tabs.

### Expected results

In Talkback mode, we should be able to swipe though 'Pizza' to 'Drink'.

### Actual results

Talkback is only able to go through the tabs initially on screen before moving on to the next widget. So the tabbar scrolls to show the other tabs, but Talkback essentially ignores them.

### Code sample

<details open><summary>Code sample</summary>

```
import 'widgets/ModTabBar';
import 'package:flutter/material.dart';

class TabBarsExample extends StatelessWidget {
  const TabBarsExample({super.key});

  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
      length: 7,
      child: Scaffold(
        appBar: AppBar(
          title: const Text('ModTabBar demo'),
          bottom: const ModTabBar(
            tabs: [
              Tab(text: 'Pizza'),
              Tab(text: 'Wings'),
              Tab(text: 'Sides'),
              Tab(text: 'Sandwiches'),
              Tab(text: 'Salads'),
              Tab(text: 'Treats'),
              Tab(text: 'Drinks'),
            ],
          ),
        ),
      ),
    );
  }
}
```

```
import 'package:flutter/material.dart';

class ModTabBar extends StatelessWidget implements PreferredSizeWidget {
  const ModTabBar({
    super.key,
    required this.tabs,
    this.isScrollable = true,
    this.showBottomDivider = true,
    this.onTap,
  });

  final List<Widget> tabs;
  final bool isScrollable;
  final bool showBottomDivider;
  final ValueChanged<int>? onTap;

  @override
  Widget build(BuildContext context) {
    return Container(
      color: Colors.white,
      child: TabBar(
        dividerHeight: showBottomDivider ? 1 : 0,
        onTap: onTap,
        isScrollable: isScrollable,
        padding: EdgeInsets.zero,
        indicatorWeight: 4,
        automaticIndicatorColorAdjustment: false,
        tabs: tabs,
      ),
    );
  }

  @override
  Size get preferredSize => const Size.fromHeight(44);
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

See this comment for a demo: https://github.com/flutter/flutter/issues/176237#issuecomment-3382290251
</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.35.2, on macOS 26.0.1 25A362 darwin-arm64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 36.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 26.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2025.1.4.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!

```

</details>

**Comment by huycozy** [comment]
Thanks for filing a separate issue, @kynara 
I see your sample code is incomplete (`ModTabBar` is found). Can this issue be reproduced with a simple [TabBar](https://api.flutter.dev/flutter/material/TabBar-class.html) widget? If so, can you provide that sample code?

**Comment by kynara** [comment]
Hi @huycozy! The code sample includes the implementation of `ModTabBar`. Here it is again, it uses the simple TabBar: 
```
import 'package:flutter/material.dart';

class ModTabBar extends StatelessWidget implements PreferredSizeWidget {
  const ModTabBar({
    super.key,
    required this.tabs,
    this.isScrollable = true,
    this.showBottomDivider = true,
    this.onTap,
  });

  final List<Widget> tabs;
  final bool isScrollable;
  final bool showBottomDivider;
  final ValueChanged<int>? onTap;

  @override
  Widget build(BuildContext context) {
    return Container(
      color: Colors.white,
      child: TabBar(
        dividerHeight: showBottomDivider ? 1 : 0,
        onTap: onTap,
        isScrollable: isScrollable,
        padding: EdgeInsets.zero,
        indicatorWeight: 4,
        automaticIndicatorColorAdjustment: false,
        tabs: tabs,
      ),
    );
  }

  @override
  Size get preferredSize => const Size.fromHeight(44);
}
```

**Comment by huycozy** [comment]
Thanks for your update, (and sorry for didn't check out all your provided code above).

I tested it on the latest Flutter stable 3.35.6 and see it works as expected. Can you upgrade Flutter SDK and retry?

https://github.com/user-attachments/assets/fc567d80-12ca-46d2-95a3-d12485145b20

**Comment by kynara** [comment]
Thank you! I am also having the same issue on 3.35.6. This helped me narrow down the issue more! @huycozy on your app, can you try adding this theme to your `MaterialApp`:
```  
theme: ThemeData(
        tabBarTheme: TabBarThemeData(
          tabAlignment: TabAlignment.center,
        ),
      ),
```

The tab alignment seems to be what is causing my issue
